### ISSUE TITLE
`IS2_DEM_difference` caches the result but doesn't fetch it again

### ISSUE THREAD

**Issue by erikmannerfelt** [eliskasieglova/SvalbardSurges#6]
It's great that you're caching the result of the DEM differencing! There's however no cache retrieval logic so the process is rerun every time!

https://github.com/eliskasieglova/SvalbardSurges/blob/448be056b27c268483c5b60abed28f32465b97db/main.py#L190-L191

This is an easy fix by simply adding something almost identical to what's in `subset_is2`:
https://github.com/eliskasieglova/SvalbardSurges/blob/448be056b27c268483c5b60abed28f32465b97db/main.py#L67-L71

Basically, make the `cache_path` at the start of the function, then add the cache check just after.
